WalletWasabi.Gui/Program.cs

```diff
@@ -48,26 +50,21 @@ private static void Main(string[] args)
 				if (Global.CrashReporter.IsReport)
 				{
 					StartCrashReporter(args);
-					return;
 				}
-
-				if (runGui)
+				else if (runGui)
 				{
 					Logger.LogSoftwareStarted("Wasabi GUI");
-
 					BuildAvaloniaApp().StartShellApp("Wasabi Wallet", AppMainAsync, args);
 				}
 			}
 			catch (Exception ex)
 			{
+				appException = ex;
 				Logger.LogCritical(ex);
-				Global.CrashReporter.SetException(ex);
-				throw;
-			}
-			finally
-			{
-				DisposeAsync().GetAwaiter().GetResult();
+				Global?.CrashReporter?.SetException(ex);
 			}
+
+			TerminateApplicationAsync(appException).GetAwaiter().GetResult();
```

Comment:
In the future, it may be good to actually remove `CrashReporter` from `Global`, I think it would make your code in `Program.cs` easier actually.

Reply:
I see you did that in #4586. Good job.

Reply:
It is removed here https://github.com/zkSNACKs/WalletWasabi/pull/4586